### PR TITLE
Add psalm as linter

### PR DESCRIPTION
**Pull request by danrot**
I copied this part of my nvim configuration. Is that everything which is needed? Or do I have to additionally register it somehow?

### CODE REVIEW

**Comment by mfussenegger** [comment]
Thanks for the PR.
Adding an entry to the table in the readme would be nice, otherwise this is all that's needed.

The linter is unhappy about indentation, other than that this looks good to merge to me.

**Comment by danrot** [comment]
All done 🙂

File: lua/lint/linters/psalm.lua [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    for _, message in ipairs(messages or {}) do
+      table.insert(diagnostics, {
+        lnum = message.line_from - 1,
+        end_lnum = message.line_to - 1,
+        col = message.column_from - 1,
+        end_col = message.column_to - 1,
+        message = message.message,
+        source = 'psalm',
+        severity = message.severity,
```
Comment (mfussenegger) [inline lua/lint/linters/psalm.lua:23]: Sorry I missed this initially, `severity` must be one of `:h diagnostic-severity`. Unless `message.severity` happens to be a number from 1-4 that happens to align 1:1 with the vim.diagnostic definition it needs to be mapped.

Other linters usually have some kind of mapping:

```
local severities = {
  error = vim.diagnostic.severity.ERROR,
  warning = vim.diagnostic.severity.WARN,
  information = vim.diagnostic.severity.INFO,
  hint = vim.diagnostic.severity.HINT,
  suggestion = vim.diagnostic.severity.HINT,
}
```

And then use something like `severities[message.severity] or vim.diagnostic.severity.WARN` to retrieve it and have a default value.

Comment (danrot) [reply lua/lint/linters/psalm.lua:23]: I have found info and error diagnostics, and the both seem to work as I have implemented it currently:

image

Comment (mfussenegger) [reply lua/lint/linters/psalm.lua:23]: Alright,  if it turns out to miss some states or map it incorrectly we can still extend it later.  Merged the PR as is

Comment (danrot) [reply lua/lint/linters/psalm.lua:23]: Thank you!

**Pull request merged**